### PR TITLE
#984 -Sort Tasks by deadline and date created

### LINKED ISSUE #984: Sort Tasks by deadline and date created
### Description

Users want to see the tasks with the closest or overdue deadlines first. If two tasks have the same due date, then sort them by the date they were created, with earlier dates being in front of later dates.

### Acceptance Criteria

Satisfies Description.

### Proposed Solution

Edit the options that are being given to the data grid to be sorted in this way.

### Mocks

_No response_

### PR DESCRIPTION
**Pull request by Lomloam**
## Changes

Sort Tasks by deadline and date created

image

image

- [x] All commits are tagged with the ticket number
- [ ] No linting errors / newline at end of file warnings
- [ ] All code follows repository-configured prettier formatting
- [ ] No merge conflicts
- [ ] All checks passing
- [ ] Screenshots of UI changes (see Screenshots section)
- [ ] Remove any non-applicable sections of this template
- [ ] Assign the PR to yourself
- [ ] No `yarn.lock` changes (unless dependencies have changed)
- [ ] Request reviewers & ping on Slack
- [ ] PR is linked to the ticket (fill in the closes line below)

Closes #984

### CODE REVIEW

**Review by Peyton-McKee — APPROVED** [review]
Yeah this looks pretty good. Good work!

**Comment by Lomloam** [comment]
> 

thanks

File: src/frontend/src/pages/ProjectDetailPage/ProjectViewContainer/TaskList.tsx [hunk 1]
```diff
@@ -24,11 +24,32 @@ const TaskList = ({ project, defaultClosed }: TaskListProps) => {
...
   const [value, setValue] = useState<number>(1);
   const [addTask, setAddTask] = useState(false);
-
   const tasks = project.tasks;
-  const backLogTasks = tasks.filter((task: Task) => task.status === TaskStatus.IN_BACKLOG);
-  const inProgressTasks = tasks.filter((task: Task) => task.status === TaskStatus.IN_PROGRESS);
-  const doneTasks = tasks.filter((task: Task) => task.status === TaskStatus.DONE);
+
+  //used two sort two Tasks based on ascending times
+  const sortAscDate = (task1: Task, task2: Task) => {
```
Comment (anthonybernardi) [inline src/frontend/src/pages/ProjectDetailPage/ProjectViewContainer/TaskList.tsx:30]: `sortAscendingDate`

File: src/frontend/src/pages/ProjectDetailPage/ProjectViewContainer/TaskList.tsx [hunk 2]
```diff
@@ -24,11 +24,32 @@ const TaskList = ({ project, defaultClosed }: TaskListProps) => {
...
+    const deadLine1 = task1.deadline.getTime();
+    const deadLine2 = task2.deadline.getTime();
+
+    if (deadLine1 < deadLine2) {
```
Comment (anthonybernardi) [inline src/frontend/src/pages/ProjectDetailPage/ProjectViewContainer/TaskList.tsx:34]:
```
if (deadline1 !== deadline2) return deadline1 - deadline2
return task1.datecreated.getTime() - task2.datecreated.gettime()
```

(make sure that this still does it in ascending order)

File: src/frontend/src/pages/ProjectDetailPage/ProjectViewContainer/TaskList.tsx [hunk 3]
```diff
@@ -24,11 +24,32 @@ const TaskList = ({ project, defaultClosed }: TaskListProps) => {
   const auth = useAuth();
   const [value, setValue] = useState<number>(1);
   const [addTask, setAddTask] = useState(false);
-
   const tasks = project.tasks;
```
Comment (anthonybernardi) [inline src/frontend/src/pages/ProjectDetailPage/ProjectViewContainer/TaskList.tsx:27]: put the sort function at the top outside of the component and then sort it here (then you dont have to call the sort 3 times below)

**Review by anthonybernardi — CHANGES_REQUESTED** [review]
nice!

File: src/frontend/src/pages/ProjectDetailPage/ProjectViewContainer/TaskList.tsx (hunk 1 above)
Comment (Lomloam) [reply src/frontend/src/pages/ProjectDetailPage/ProjectViewContainer/TaskList.tsx]: done

File: src/frontend/src/pages/ProjectDetailPage/ProjectViewContainer/TaskList.tsx (hunk 2 above)
Comment (Lomloam) [reply src/frontend/src/pages/ProjectDetailPage/ProjectViewContainer/TaskList.tsx]: done

File: src/frontend/src/pages/ProjectDetailPage/ProjectViewContainer/TaskList.tsx (hunk 3 above)
Comment (Lomloam) [reply src/frontend/src/pages/ProjectDetailPage/ProjectViewContainer/TaskList.tsx]: done

**Review by Peyton-McKee — APPROVED** [review]
lgtm

**Pull request merged**